File: DotNet/CesiumLanguageWriter/Advanced/CesiumFontValuePropertyAdaptor.cs
using JetBrains.Annotations;

namespace CesiumLanguageWriter.Advanced
{
    /// <summary>
    /// Adapts a class that implements <see cref="ICesiumFontValuePropertyWriter"/> to implement
    /// <see cref="ICesiumInterpolatableValuePropertyWriter{TValue}"/> for <see cref="string"/> values.
    /// </summary>
    /// <typeparam name="TFrom">The class that implements <see cref="ICesiumFontValuePropertyWriter"/> to adapt.</typeparam>
    public class CesiumFontValuePropertyAdaptor<TFrom> : CesiumWriterAdaptor<TFrom, string>
        where TFrom : class, ICesiumFontValuePropertyWriter
    {
        /// <summary>
        /// Initializes a new instance.
        /// </summary>
        /// <param name="parent">The instance to wrap.</param>
        /// <param name="writeValueCallback">The callback to write values of type <see cref="string"/>.</param>
        /// <param name="writeDeleteValueCallback">The callback to write an indication that the client should delete existing data.</param>
        public CesiumFontValuePropertyAdaptor([NotNull] TFrom parent,
                                              [NotNull] CesiumWriterAdaptorWriteCallback<TFrom, string> writeValueCallback,
                                              [NotNull] CesiumWriterAdaptorWriteDeleteCallback<TFrom> writeDeleteValueCallback)
            : base(parent, writeValueCallback, writeDeleteValueCallback)
        {
        }
    }
}

File: Java/CesiumLanguageWriter/translatedSrc/cesiumlanguagewriter/advanced/CesiumFontValuePropertyAdaptor.java
package cesiumlanguagewriter.advanced;


import agi.foundation.compatibility.*;
import cesiumlanguagewriter.*;
import javax.annotation.Nonnull;

/**
 * Adapts a class that implements {@link ICesiumFontValuePropertyWriter} to implement
 {@link ICesiumInterpolatableValuePropertyWriter} for {@link String} values.
 * @param <TFrom> The class that implements {@link ICesiumFontValuePropertyWriter} to adapt.
 */
@SuppressWarnings( {
        "unused",
        "deprecation",
        "serial"
})
public class CesiumFontValuePropertyAdaptor<TFrom extends ICesiumFontValuePropertyWriter> extends CesiumWriterAdaptor<TFrom, String> {
    /**
    * Initializes a new instance.
    * @param parent The instance to wrap.
    * @param writeValueCallback The callback to write values of type {@link String}.
    * @param writeDeleteValueCallback The callback to write an indication that the client should delete existing data.
    */
    public CesiumFontValuePropertyAdaptor(@Nonnull TFrom parent, @Nonnull CesiumWriterAdaptorWriteCallback<TFrom, String> writeValueCallback,
            @Nonnull CesiumWriterAdaptorWriteDeleteCallback<TFrom> writeDeleteValueCallback) {
        super(parent, writeValueCallback, writeDeleteValueCallback);
    }
}